### PR TITLE
Add support for Boost and TBB in non-standard locations

### PR DESCRIPTION
**Pull request by fwyzard**

### CODE REVIEW

**Comment by fwyzard** [comment]
Some more clean up of the `Makeifle`, to ease having libraries installed in different locations

File: Makefile [hunk 1]
```diff
@@ -2,34 +2,58 @@ TARGETS = naive cuda cupla kokkos oneapi
...
-GREEN := '\033[32m'
-RED := '\033[31m'
-RESET := '\033[0m'
+# TBB flags
+ifdef TBB_BASE
+TBB_CXX_FLAGS := -I$(TBB_BASE)/include
+TBB_LD_FLAGS  := -L$(TBB_BASE)/lib -ltbb -lrt
+else
+TBB_CXX_FLAGS :=
+TBB_LD_FLAGS  := -ltbb -lrt
```
Comment (makortel) [inline Makefile:43]: Is the assumption that if `TBB_BASE` is not defined, TBB is in a "standard location" and thus `-I$(TBB_BASE)/include` and `-L$(TBB_BASE)/lib` do not need to be specified?

**Comment by fwyzard** [comment]
Exactly, if one uses the TBB installation from the OS, we don't need to
pass -I/usr/include and -L/usr/lib .

**Comment by makortel** [comment]
Thanks

**Pull request merged**